frontend/src/store/modules/user.ts

```diff
@@ -14,7 +14,10 @@ export default {
   namespaced: true,
   state: {
     currentUser: null,
-    roles: null
+    roles: null,
+    metadataCallback: null,
+    metadataRef: null,
```

Comment:
I'm sure there is no need keeping this as a property, as we can construct the ref using `currentUser.uid`.

Reply:
At least it can become a getter.